File: CPP/Assignment6/Program4.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////
//
// File name :   Program4.cpp
// Description : Write recursive program which accept number from user & return its factorial.
// Author :      Pallavi Omprakash Malewar
// Date :        27/07/2026
//
//////////////////////////////////////////////////////////////////////////////////////////////////////

/*

    Input  :    5
    Output :    120

*/


#include<iostream>
using namespace std;

int Fact(int iNo)
{
    if(iNo == 0 || iNo == 1) return 1;
    return iNo * Fact(iNo - 1);
}

int main()
{
    int iRet = 0, iValue = 0;

    cout << "Enter Number : ";
    cin >> iValue;

    iRet = Fact(iValue);
    cout << iRet << endl;

    return 0;
}

File: JAVA/Assignment 9/Program4.java
/////////////////////////////////////////////////////////////////////////////////////////
//                                                                                     //
// Problem Statement : Write Java program which accept array of characters from user & //
//                     return difference between frequency of capital & frequency of   //
//                     small characters.                                               //
// Author : Pallavi Omprakash Malewar                                                  //
// Date : 3 Aug 2026                                                                   //
//                                                                                     //
/////////////////////////////////////////////////////////////////////////////////////////


/*  

    Input : b   N   e   B   R   b   A   I   O   G   i

    Output: 3 (7 - 4)

*/


import java.util.Scanner;

class MyArray
{
    public int Difference(char Arr[])
    {
        int n = Arr.length;
        int i = 0, smallCount = 0, capCount = 0;

        for(i = 0 ; i < n ; i++)
        {
            if(Arr[i] <= 'Z' && Arr[i] >= 'A') smallCount++;
            if(Arr[i] <= 'z' && Arr[i] >= 'a') capCount++;
        }
        return (smallCount - capCount);
    }
}

class Program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter the Size of Array : ");
        int No = sobj.nextInt();

        char Arr[] = new char[No];

        for(int i = 0 ; i < No ; i++)
        {
            Arr[i] = sobj.next().charAt(0);
        }
 
        MyArray mobj = new MyArray();

        int iRet = 0;

        iRet = mobj.Difference(Arr);

        System.out.println("Difference between frequency of capital & small characters : " + iRet);
    }
}
